### PR TITLE
Refactor train cfg cli

### PR DESCRIPTION
**Pull request by winglian**
Splitting out the non-training specific arguments into its own arguments dataclass. 
making the train function only accept pre-parsed config object so it can be called as a standalone option later on

### CODE REVIEW

File: scripts/finetune.py [hunk 1]
```diff
@@ -61,6 +77,8 @@ def get_multi_line_input() -> Optional[str]:
 
 
 def do_inference(cfg, model, tokenizer, prompter: Optional[str]):
+    if prompter == "None":
+        prompter = None
```
Comment (tmm1) [inline scripts/finetune.py:81]: Why is this needed?

**Review by tmm1 — APPROVED** [review]
Great idea!

File: scripts/finetune.py (hunk 1 above)
Comment (winglian) [reply scripts/finetune.py:81]: we have in the existing docs --prompter=None which I believe translates to a literal string. we could alternatively update the doc to be simply `--prompter` or alternatively default the value to None (which is what we're doing here)
